### PR TITLE
fix: clone server context

### PR DESCRIPTION
**Pull request by zllkjc**
# PR Details

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: pnpm-workspace.yaml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 packages:
+ - 'examples/*'
```
Comment (leeight) [inline pnpm-workspace.yaml:2]: 这个是干啥的

File: packages/server/plugin-nest/tests/function.test.ts [hunk 2]
```diff
@@ -59,9 +57,9 @@ describe('function-mode', () => {
     expect(res.body.id).toBe(777);
   });
 
-  // TODO: 后续修复（目前在 btsm + esbuild 的时候无法正常运行，缺少 ReflectMeta 的支持）
-  xit('should works with middleware', async () => {
-    const res = await request(apiHandler).get('/cats');
-    expect(res.status).toBe(200);
-  });
+  // // TODO: 后续修复（目前在 btsm + esbuild 的时候无法正常运行，缺少 ReflectMeta 的支持）
```
Comment (leeight) [inline packages/server/plugin-nest/tests/function.test.ts:60]: 这个可以留着吧？

File: pnpm-workspace.yaml (hunk 1 above)
Comment (zllkjc) [reply pnpm-workspace.yaml]: 删了~

File: packages/server/plugin-nest/tests/function.test.ts (hunk 2 above)
Comment (zllkjc) [reply packages/server/plugin-nest/tests/function.test.ts:61]: OK

**Review by yimingjfe — APPROVED** [review]

**Pull request merged**